src/engine.spec.ts

```diff
@@ -632,348 +632,392 @@ describe("Engine", () => {
       });
   });
 
-  describe("free actions", () => {
-    it("should allow free action as first move after setupsetup is correct", () => {
+  describe('Baltaks', () => {
+    it('should test ability to upgrade navigation', () => {
       const moves = parseMoves(`
         init 2 randomSeed
         p1 faction terrans
-        p2 faction nevlas
-        p1 build m -4x-1
-        p2 build m -1x0
-        p2 build m 0x-4
+        p2 faction baltaks
         p1 build m -4x2
-        p2 booster booster3
-        p1 booster booster4
-        p1 spend 1q for 1o
+        p2 build m -4x0
+        p2 build m -1x-3
+        p1 build m -1x2
+        p2 booster booster4
+        p1 booster booster3
+        p1 pass booster5
+        p2 up nav.
       `);
 
-      expect(() => new Engine(moves)).to.not.throw();
+      expect(() => new Engine(moves)).to.throw();
     });
 
-    it("should allow free actions to spend 1q for 1c", () => {
+    it.only('should test ability to use freeaction and reset on new turn', () => {
       const moves = parseMoves(`
         init 2 randomSeed
         p1 faction terrans
-        p2 faction nevlas
-        p1 build m -4x-1
-        p2 build m -1x0
-        p2 build m 0x-4
+        p2 faction baltaks
         p1 build m -4x2
-        p2 booster booster3
-        p1 booster booster4
-        p1 spend 1q for 1c
+        p2 build m -4x0
+        p2 build m -1x-3
+        p1 build m -1x2
+        p2 booster booster4
+        p1 booster booster3
+        p1 pass booster5
+        p2 spend 1gf for 1q. build m -3x-2.
       `);
+      const engine = new Engine(moves);
+
+
+      expect(engine.player(Player.Player2).data.gaiaformersInGaia).to.equal(1);
+      engine.move("p2 pass booster3");
+      expect(engine.player(Player.Player2).data.gaiaformersInGaia).to.equal(0);
 
-      expect(() => new Engine(moves)).to.not.throw();
     });
 
-    it("should prevent unreasonable free actions", () => {
-      const moves = parseMoves(`
-        init 2 randomSeed
-        p1 faction terrans
-        p2 faction nevlas
-        p1 build m -4x-1
-        p2 build m -1x0
-        p2 build m 0x-4
-        p1 build m -4x2
-        p2 booster booster3
-        p1 booster booster4
-      `);
+  });
 
-      const engine = new Engine(moves);
+  describe("free actions", () => {
+  it("should allow free action as first move after setupsetup is correct", () => {
```

Comment:
The indent is broken. It causes a huge commit !!

Reply:
I'll resubmit one with the right indentation. Why isn't lint correcting this?

Reply:
I've only added the Baltaks part (describe("Baltaks")

Reply:
Working on the tests... please wait